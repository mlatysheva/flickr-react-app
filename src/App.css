* {
  box-sizing: border-box;
}

.App {
  width: 100%;
  height: 100%;
  margin: 0 auto;
  text-align: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.header {
  width: 100%;
  height: 80px;
  padding: 1rem;
  background-color: midnightblue;
  color: white;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
}

.header a {
  color: white;
}

.header a.active {
  color: #87ceeb;
}

.theme-selector {
  width: 3rem;
  height: 3rem;
  margin-right: 2rem;
  background-image: url(./assets/dark_light_theme.svg);
  background-size: contain;
  background-position: center;
  background-repeat: no-repeat;
  filter: invert(100%) sepia(0%) saturate(0%) hue-rotate(154deg) brightness(105%) contrast(102%);
  cursor: pointer;
}
.theme-dark {
  filter: invert(68%) sepia(89%) saturate(182%) hue-rotate(156deg) brightness(97%) contrast(89%);
}

.current-page-title {
  margin-right: auto;
  margin-left: 5rem;
  font-weight: 700;
  text-transform: uppercase;
}

.here-text {
  font-weight: 400;
  text-transform: none;
}

.nav {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  margin-right: 2rem;
}

.main {
  max-width: 1440px;
  min-height: calc(100vh - 160px);
  flex: 1;
  display: flex;
  flex-direction: column;
  margin: 0 auto;
  align-items: center;
}

.search {
  margin: 1rem auto;
  padding: 1rem;
  width: 70%;
  display: flex;
  flex-direction: row;
  align-items: center;
  border-radius: 10px;
  box-shadow: 0px 5px 10px 2px rgba(34, 60, 80, 0.2);
}

.search-parameter {
  width: auto;
  min-width: 200px;
}

.search-button {
  display: block;
  margin-right: 3rem;
  margin-left: 4rem;
}

.search-field {
  width: 400px;
  height: 1.5rem;
  margin-right: auto;
  border: 1px solid grey;
  border-radius: 5px;
  padding-left: 1rem;
}

.cards-wrapper {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

.card {
  width: 350px;
  display: flex;
  flex-direction: column;
  padding: 1rem;
  margin: 1rem;
  border-radius: 10px;
  box-shadow: 0px 5px 10px 2px rgba(34, 60, 80, 0.2);
  cursor: pointer;
}

.title-wrapper {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
}

.cards-wrapper a {
  color: midnightblue;
}

.cards-wrapper a:visited {
  color: #87ceeb;
}

.country {
  font-size: larger;
  font-weight: 600;
}

.flag-image {
  width: 64px;
  height: 40px;
  margin-left: 1rem;
  margin-right: 2rem;
  box-shadow: 0px 5px 10px 2px rgba(34, 60, 80, 0.2);
}

.explanation {
  font-style: italic;
}

.main-form {
  width: 100%;
  margin: 1rem auto;
  display: flex;
  flex-direction: column;
}

.form-title {
  font-size: x-large;
  padding-right: 1rem;
}

.form-container {
  width: 100%;
  margin: 1rem auto;
  display: flex;
  text-align: left;
  flex-direction: column;
  align-items: center;
}

fieldset { 
  border: none;
  width: 100%;
  max-width: 600px;
  margin-top: 20px;
  border-radius: 10px;
  box-shadow: 0px 5px 10px 2px rgba(34, 60, 80, 0.2);
  padding: 1rem; 
} 

legend { 
  padding: 1rem; 
  font-size: x-large;
}

label { 
  color: black; 
  font-size: large; 
  display: block; 
  width: 200px; 
  float: left; 
} 

.form-input-field {
  margin-right: 2rem;
  max-width: 400px;
  height: 1.5rem;
  width: 300px;
  border: 1px solid grey;
  border-radius: 5px;
}

.checkbox-field {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid grey;
  border-radius: 4px;
  outline: none;
  transition-duration: 0.3s;
  transition-property: all;
  cursor: pointer;
  height: 20px;
  width: 20px;
}

.checkbox-field::before {
  content: "\2713";
  color: white;
  padding-left: 2px;
  font-size: 16px;
  text-align: center;
}

.checkbox-field:checked {
  background-color: midnightblue;
}

.subtitle {
  font-size: large;
}

.smaller-font {
  font-size: medium;
}

.radio-container {
  display: flex;
  flex-direction: row;
}

.radio-wrapper {
  display: flex;
  flex-direction: row;
  margin-bottom: 5px;
}

.radio-checkbox {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  margin-right: 1rem;
  width: 20px;
  height: 20px;  
  padding: 3px;
  background-clip: content-box;
  border: 2px solid #060c3b;
  border-radius: 50%;
}

.radio-checkbox:checked {
  background-color: midnightblue;
}

.radio-checkbox:hover {
  box-shadow: 0 0 5px 0px blue inset;
}

.file-upload-field {
  border: none;
}

.submit-form-button {
  margin-top: 2rem;
  align-self: center;
}

.error-message {
  color: red;
  display: block;
  margin-left: 200px;
}

.input-amount {
  width: 150px;
  height: 40px;
  margin: 1rem;
  text-align: center;
  display: block;
  font-size: large;
  border-radius: 5px;
  border: 1px solid midnightblue;
  box-shadow: 0px 5px 10px 2px rgb(34 60 80 / 20%);
  /* cursor: pointer; */
}

.footer {
  width: 100%;
  height: 80px;
  padding: 1rem;
  bottom: 0;
  background-color: midnightblue;
  color: white;  
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
}

.logo {
  width: 100px;
  background: rgba(255, 255, 255, 0.3);
  margin-left: 1rem;
}

.footer a {
  color:white;
  display: block;
  text-transform: none;
}

.photo-image {
  margin: 1rem;
  margin-right: 2rem;
  box-shadow: 0px 5px 10px 2px rgba(34, 60, 80, 0.2);
}

.api-card-link {
  height: 2rem;
  background-color: transparent;
  color:mediumblue;
  margin-top: auto;
  border: none;
  align-self: center;
}

.api-card-link:hover {
  color: cornflowerblue;
}

.small-api-card {
  height: 200px;
  align-items:flex-start;
}

@media (max-width: 1400px) {
  
  .search-parameter {
    justify-content: center;

  }
  .form-input-field {
    width: auto;
    margin-top: 0.5rem;
  }
}
@media (max-width: 770px) {
  .search {
    flex-wrap: wrap;
    flex-direction: column;
  }
}

@media (max-width: 1050px) {
  .search {
    width: 90%;
  }
}
