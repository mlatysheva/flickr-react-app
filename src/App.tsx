import { Routes, Route } from 'react-router-dom';
import { store } from './store';
import Home from './components/Home';
import About from './components/About';
import Error from './components/Error';
import './App.css';
import Header from './components/Header';
import ErrorBoundary from './components/ErrorBoundary';
import Footer from './components/Footer';
import { FormPage } from './components/FormPage';
import { ApiPage } from './components/ApiPage';
import { ThemeProvider } from './components/Theme';
import ApiPhoto from './components/ApiPhoto';
import { Provider } from 'react-redux';

function App() {
  return (
    <div className="App">
      <ErrorBoundary>
        <Provider store={store}>
          <ThemeProvider>
            <Header />
            <Routes>
              <Route path="/" element={<Home />} />
              <Route path="about" element={<About />} />
              <Route path="form" element={<FormPage />} />
              <Route path="api" element={<ApiPage />} />
              <Route path="api/photo" element={<ApiPhoto />} />
              <Route path="error" element={<Error />} />
              <Route path="*" element={<Error />} />
            </Routes>
            <Footer />
          </ThemeProvider>
        </Provider>
      </ErrorBoundary>
    </div>
  );
}

export default App;
