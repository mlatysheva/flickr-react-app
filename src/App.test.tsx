import { render, screen } from '@testing-library/react';
import App from './App';
import { MemoryRouter } from 'react-router-dom';

test('renders text', () => {
  render(
    <MemoryRouter>
      <App />
    </MemoryRouter>);
  const text = screen.getByText(/G20 Countries/i);
  expect(text).toBeInTheDocument();
});

test('renders navigation link', () => {
  render(
    <MemoryRouter>
      <App />
    </MemoryRouter>);
  const linkElement = screen.getByText(/ABOUT/i);
  expect(linkElement).toBeInTheDocument();
});

test('content of App matches the snapshot', () => {
  const { asFragment } = render(
    <MemoryRouter>
      <App />
    </MemoryRouter>);
  expect(asFragment()).toMatchSnapshot();
});

describe('textbox role is in the document', () => {
  it('renders App component', () => {
    render(<MemoryRouter>
      <App />
    </MemoryRouter>);
    expect(screen.getByRole('textbox')).toBeInTheDocument();
  })
})