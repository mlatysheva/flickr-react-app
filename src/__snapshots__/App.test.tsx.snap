// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`content of App matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="header"
    >
      <div
        class="current-page-title"
      >
        <span
          class="here-text"
        >
          Current page: 
        </span>
        Home
      </div>
      <nav
        class="nav"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          Home
        </a>
        <a
          class=""
          href="/form"
        >
          Form
        </a>
        <a
          class=""
          href="/api"
        >
          API
        </a>
        <a
          class=""
          href="/about"
        >
          About
        </a>
      </nav>
      <div
        class="theme-selector theme-dark"
      />
    </header>
    <div
      class="main"
    >
      <h1>
        G20 Countries
      </h1>
      <p>
        Filter G20 members by continent:
      </p>
      <form
        class="search"
      >
        <input
          class="search-field"
          id="search"
          name="search"
          placeholder="Enter continent"
          type="text"
          value=""
        />
        <button
          class="search-button"
          style="background: rgb(238, 238, 238); color: rgb(0, 0, 0);"
        >
          Search
        </button>
      </form>
      <div
        class="cards-wrapper"
      >
        <div
          class="card"
        >
          <div
            class="title-wrapper"
          >
            <img
              alt="flag"
              class="flag-image"
              src="Flag_of_United_States.png"
            />
            <p
              class="country"
            >
              United States
            </p>
          </div>
          <p
            class="continent"
          >
            <span
              class="explanation"
            >
              Continent: 
            </span>
            North America
          </p>
          <p
            class="population"
          >
            <span
              class="explanation"
            >
              Population, mln: 
            </span>
            334.8
          </p>
          <p
            class="gdp"
          >
            <span
              class="explanation"
            >
              GDP, $bln: 
            </span>
            21000
          </p>
        </div>
        <div
          class="card"
        >
          <div
            class="title-wrapper"
          >
            <img
              alt="flag"
              class="flag-image"
              src="Flag_of_China.png"
            />
            <p
              class="country"
            >
              China
            </p>
          </div>
          <p
            class="continent"
          >
            <span
              class="explanation"
            >
              Continent: 
            </span>
            Asia
          </p>
          <p
            class="population"
          >
            <span
              class="explanation"
            >
              Population, mln: 
            </span>
            1448.5
          </p>
          <p
            class="gdp"
          >
            <span
              class="explanation"
            >
              GDP, $bln: 
            </span>
            15000
          </p>
        </div>
        <div
          class="card"
        >
          <div
            class="title-wrapper"
          >
            <img
              alt="flag"
              class="flag-image"
              src="Flag_of_Europe.png"
            />
            <p
              class="country"
            >
              Eurean Union
            </p>
          </div>
          <p
            class="continent"
          >
            <span
              class="explanation"
            >
              Continent: 
            </span>
            Europe
          </p>
          <p
            class="population"
          >
            <span
              class="explanation"
            >
              Population, mln: 
            </span>
            450
          </p>
          <p
            class="gdp"
          >
            <span
              class="explanation"
            >
              GDP, $bln: 
            </span>
            13010
          </p>
        </div>
        <div
          class="card"
        >
          <div
            class="title-wrapper"
          >
            <img
              alt="flag"
              class="flag-image"
              src="Flag_of_Japan.png"
            />
            <p
              class="country"
            >
              Japan
            </p>
          </div>
          <p
            class="continent"
          >
            <span
              class="explanation"
            >
              Continent: 
            </span>
            Asia
          </p>
          <p
            class="population"
          >
            <span
              class="explanation"
            >
              Population, mln: 
            </span>
            125.6
          </p>
          <p
            class="gdp"
          >
            <span
              class="explanation"
            >
              GDP, $bln: 
            </span>
            5000
          </p>
        </div>
        <div
          class="card"
        >
          <div
            class="title-wrapper"
          >
            <img
              alt="flag"
              class="flag-image"
              src="Flag_of_Germany.png"
            />
            <p
              class="country"
            >
              Germany
            </p>
          </div>
          <p
            class="continent"
          >
            <span
              class="explanation"
            >
              Continent: 
            </span>
            Europe
          </p>
          <p
            class="population"
          >
            <span
              class="explanation"
            >
              Population, mln: 
            </span>
            83.8
          </p>
          <p
            class="gdp"
          >
            <span
              class="explanation"
            >
              GDP, $bln: 
            </span>
            3850
          </p>
        </div>
        <div
          class="card"
        >
          <div
            class="title-wrapper"
          >
            <img
              alt="flag"
              class="flag-image"
              src="Flag_of_United_Kingdom.png"
            />
            <p
              class="country"
            >
              United Kingdom
            </p>
          </div>
          <p
            class="continent"
          >
            <span
              class="explanation"
            >
              Continent: 
            </span>
            Europe
          </p>
          <p
            class="population"
          >
            <span
              class="explanation"
            >
              Population, mln: 
            </span>
            68.5
          </p>
          <p
            class="gdp"
          >
            <span
              class="explanation"
            >
              GDP, $bln: 
            </span>
            2710
          </p>
        </div>
        <div
          class="card"
        >
          <div
            class="title-wrapper"
          >
            <img
              alt="flag"
              class="flag-image"
              src="Flag_of_France.png"
            />
            <p
              class="country"
            >
              France
            </p>
          </div>
          <p
            class="continent"
          >
            <span
              class="explanation"
            >
              Continent: 
            </span>
            Europe
          </p>
          <p
            class="population"
          >
            <span
              class="explanation"
            >
              Population, mln: 
            </span>
            65.6
          </p>
          <p
            class="gdp"
          >
            <span
              class="explanation"
            >
              GDP, $bln: 
            </span>
            2630
          </p>
        </div>
        <div
          class="card"
        >
          <div
            class="title-wrapper"
          >
            <img
              alt="flag"
              class="flag-image"
              src="Flag_of_India.png"
            />
            <p
              class="country"
            >
              India
            </p>
          </div>
          <p
            class="continent"
          >
            <span
              class="explanation"
            >
              Continent: 
            </span>
            Asia
          </p>
          <p
            class="population"
          >
            <span
              class="explanation"
            >
              Population, mln: 
            </span>
            1406.6
          </p>
          <p
            class="gdp"
          >
            <span
              class="explanation"
            >
              GDP, $bln: 
            </span>
            2600
          </p>
        </div>
        <div
          class="card"
        >
          <div
            class="title-wrapper"
          >
            <img
              alt="flag"
              class="flag-image"
              src="Flag_of_Italy.png"
            />
            <p
              class="country"
            >
              Italy
            </p>
          </div>
          <p
            class="continent"
          >
            <span
              class="explanation"
            >
              Continent: 
            </span>
            Europe
          </p>
          <p
            class="population"
          >
            <span
              class="explanation"
            >
              Population, mln: 
            </span>
            60.3
          </p>
          <p
            class="gdp"
          >
            <span
              class="explanation"
            >
              GDP, $bln: 
            </span>
            1900
          </p>
        </div>
        <div
          class="card"
        >
          <div
            class="title-wrapper"
          >
            <img
              alt="flag"
              class="flag-image"
              src="Flag_of_Canada.png"
            />
            <p
              class="country"
            >
              Canada
            </p>
          </div>
          <p
            class="continent"
          >
            <span
              class="explanation"
            >
              Continent: 
            </span>
            North America
          </p>
          <p
            class="population"
          >
            <span
              class="explanation"
            >
              Population, mln: 
            </span>
            38.4
          </p>
          <p
            class="gdp"
          >
            <span
              class="explanation"
            >
              GDP, $bln: 
            </span>
            1650
          </p>
        </div>
        <div
          class="card"
        >
          <div
            class="title-wrapper"
          >
            <img
              alt="flag"
              class="flag-image"
              src="Flag_of_South_Korea.png"
            />
            <p
              class="country"
            >
              South Korea
            </p>
          </div>
          <p
            class="continent"
          >
            <span
              class="explanation"
            >
              Continent: 
            </span>
            Asia
          </p>
          <p
            class="population"
          >
            <span
              class="explanation"
            >
              Population, mln: 
            </span>
            51.3
          </p>
          <p
            class="gdp"
          >
            <span
              class="explanation"
            >
              GDP, $bln: 
            </span>
            1630
          </p>
        </div>
        <div
          class="card"
        >
          <div
            class="title-wrapper"
          >
            <img
              alt="flag"
              class="flag-image"
              src="Flag_of_Russia.png"
            />
            <p
              class="country"
            >
              Russia
            </p>
          </div>
          <p
            class="continent"
          >
            <span
              class="explanation"
            >
              Continent: 
            </span>
            Europe
          </p>
          <p
            class="population"
          >
            <span
              class="explanation"
            >
              Population, mln: 
            </span>
            145.8
          </p>
          <p
            class="gdp"
          >
            <span
              class="explanation"
            >
              GDP, $bln: 
            </span>
            1480
          </p>
        </div>
        <div
          class="card"
        >
          <div
            class="title-wrapper"
          >
            <img
              alt="flag"
              class="flag-image"
              src="Flag_of_Brazil.png"
            />
            <p
              class="country"
            >
              Brazil
            </p>
          </div>
          <p
            class="continent"
          >
            <span
              class="explanation"
            >
              Continent: 
            </span>
            South America
          </p>
          <p
            class="population"
          >
            <span
              class="explanation"
            >
              Population, mln: 
            </span>
            215.3
          </p>
          <p
            class="gdp"
          >
            <span
              class="explanation"
            >
              GDP, $bln: 
            </span>
            1445
          </p>
        </div>
        <div
          class="card"
        >
          <div
            class="title-wrapper"
          >
            <img
              alt="flag"
              class="flag-image"
              src="Flag_of_Australia.png"
            />
            <p
              class="country"
            >
              Australia
            </p>
          </div>
          <p
            class="continent"
          >
            <span
              class="explanation"
            >
              Continent: 
            </span>
            Australia
          </p>
          <p
            class="population"
          >
            <span
              class="explanation"
            >
              Population, mln: 
            </span>
            26.07
          </p>
          <p
            class="gdp"
          >
            <span
              class="explanation"
            >
              GDP, $bln: 
            </span>
            1330
          </p>
        </div>
        <div
          class="card"
        >
          <div
            class="title-wrapper"
          >
            <img
              alt="flag"
              class="flag-image"
              src="Flag_of_Spain.png"
            />
            <p
              class="country"
            >
              Spain
            </p>
          </div>
          <p
            class="continent"
          >
            <span
              class="explanation"
            >
              Continent: 
            </span>
            Europe
          </p>
          <p
            class="population"
          >
            <span
              class="explanation"
            >
              Population, mln: 
            </span>
            50
          </p>
          <p
            class="gdp"
          >
            <span
              class="explanation"
            >
              GDP, $bln: 
            </span>
            1280
          </p>
        </div>
        <div
          class="card"
        >
          <div
            class="title-wrapper"
          >
            <img
              alt="flag"
              class="flag-image"
              src="Flag_of_Mexico.png"
            />
            <p
              class="country"
            >
              Mexico
            </p>
          </div>
          <p
            class="continent"
          >
            <span
              class="explanation"
            >
              Continent: 
            </span>
            North America
          </p>
          <p
            class="population"
          >
            <span
              class="explanation"
            >
              Population, mln: 
            </span>
            131.6
          </p>
          <p
            class="gdp"
          >
            <span
              class="explanation"
            >
              GDP, $bln: 
            </span>
            1080
          </p>
        </div>
        <div
          class="card"
        >
          <div
            class="title-wrapper"
          >
            <img
              alt="flag"
              class="flag-image"
              src="Flag_of_Indonesia.png"
            />
            <p
              class="country"
            >
              Indonesia
            </p>
          </div>
          <p
            class="continent"
          >
            <span
              class="explanation"
            >
              Continent: 
            </span>
            Asia
          </p>
          <p
            class="population"
          >
            <span
              class="explanation"
            >
              Population, mln: 
            </span>
            279.1
          </p>
          <p
            class="gdp"
          >
            <span
              class="explanation"
            >
              GDP, $bln: 
            </span>
            1060
          </p>
        </div>
        <div
          class="card"
        >
          <div
            class="title-wrapper"
          >
            <img
              alt="flag"
              class="flag-image"
              src="Flag_of_Switzerland.png"
            />
            <p
              class="country"
            >
              Switzerland
            </p>
          </div>
          <p
            class="continent"
          >
            <span
              class="explanation"
            >
              Continent: 
            </span>
            Europe
          </p>
          <p
            class="population"
          >
            <span
              class="explanation"
            >
              Population, mln: 
            </span>
            15
          </p>
          <p
            class="gdp"
          >
            <span
              class="explanation"
            >
              GDP, $bln: 
            </span>
            750
          </p>
        </div>
        <div
          class="card"
        >
          <div
            class="title-wrapper"
          >
            <img
              alt="flag"
              class="flag-image"
              src="Flag_of_Turkey.png"
            />
            <p
              class="country"
            >
              Turkey
            </p>
          </div>
          <p
            class="continent"
          >
            <span
              class="explanation"
            >
              Continent: 
            </span>
            Europe
          </p>
          <p
            class="population"
          >
            <span
              class="explanation"
            >
              Population, mln: 
            </span>
            85.6
          </p>
          <p
            class="gdp"
          >
            <span
              class="explanation"
            >
              GDP, $bln: 
            </span>
            720
          </p>
        </div>
        <div
          class="card"
        >
          <div
            class="title-wrapper"
          >
            <img
              alt="flag"
              class="flag-image"
              src="Flag_of_Saudi_Arabia.png"
            />
            <p
              class="country"
            >
              Saudi Arabia
            </p>
          </div>
          <p
            class="continent"
          >
            <span
              class="explanation"
            >
              Continent: 
            </span>
            Asia
          </p>
          <p
            class="population"
          >
            <span
              class="explanation"
            >
              Population, mln: 
            </span>
            35.8
          </p>
          <p
            class="gdp"
          >
            <span
              class="explanation"
            >
              GDP, $bln: 
            </span>
            700
          </p>
        </div>
        <div
          class="card"
        >
          <div
            class="title-wrapper"
          >
            <img
              alt="flag"
              class="flag-image"
              src="Flag_of_Argentina.png"
            />
            <p
              class="country"
            >
              Argentina
            </p>
          </div>
          <p
            class="continent"
          >
            <span
              class="explanation"
            >
              Continent: 
            </span>
            South America
          </p>
          <p
            class="population"
          >
            <span
              class="explanation"
            >
              Population, mln: 
            </span>
            46
          </p>
          <p
            class="gdp"
          >
            <span
              class="explanation"
            >
              GDP, $bln: 
            </span>
            380
          </p>
        </div>
        <div
          class="card"
        >
          <div
            class="title-wrapper"
          >
            <img
              alt="flag"
              class="flag-image"
              src="Flag_of_Singapore.png"
            />
            <p
              class="country"
            >
              Singapore
            </p>
          </div>
          <p
            class="continent"
          >
            <span
              class="explanation"
            >
              Continent: 
            </span>
            Asia
          </p>
          <p
            class="population"
          >
            <span
              class="explanation"
            >
              Population, mln: 
            </span>
            25
          </p>
          <p
            class="gdp"
          >
            <span
              class="explanation"
            >
              GDP, $bln: 
            </span>
            340
          </p>
        </div>
        <div
          class="card"
        >
          <div
            class="title-wrapper"
          >
            <img
              alt="flag"
              class="flag-image"
              src="Flag_of_South_Africa.png"
            />
            <p
              class="country"
            >
              South Africa
            </p>
          </div>
          <p
            class="continent"
          >
            <span
              class="explanation"
            >
              Continent: 
            </span>
            Africa
          </p>
          <p
            class="population"
          >
            <span
              class="explanation"
            >
              Population, mln: 
            </span>
            43
          </p>
          <p
            class="gdp"
          >
            <span
              class="explanation"
            >
              GDP, $bln: 
            </span>
            300
          </p>
        </div>
      </div>
    </div>
    <footer
      class="footer"
    >
      <img
        alt="logo of RS School"
        class="logo"
        src="https://rs.school/images/rs_school_js.svg"
      />
      <a
        href="mailto:mlatysheva@gmail.com"
      >
        Maria Latysheva © 2022
      </a>
      <a
        class="rsschool"
        href="https://rs.school"
      >
        The Rolling Slopes School
      </a>
      <a
        class="mygit"
        href="https://github.com/mlatysheva"
      >
        mlatysheva
      </a>
    </footer>
  </div>
</DocumentFragment>
`;
